private/rules/maven_utils.bzl

```diff
@@ -1,10 +1,10 @@
 def unpack_coordinates(coords):
     """Takes a maven coordinate and unpacks it into a struct with fields
-    `groupId`, `artifactId`, `version`, `type`, `scope`
+    `groupId`, `artifactId`, `version`, `type`, `classifier`
```

Comment:
> since I know that are plenty of folks out there who use this format for their test scoped dependencies

Can you provide an example of how `unpack_coordinates` is used in those cases?

I do think `groupId:artifactId[:type[:classifier]]:version` is the "correct" way to do things, but if there's concerns about keeping things compatible, I'd like to see how its being used. I think I covered all the use cases *within* the `rules_jvm_external` repo.

Reply:
I've got a version of `unpack_coordinates` that preserves the existing format and also supports the Gradle "external dependency" format [in this PR](https://github.com/bazelbuild/rules_jvm_external/pull/1232/files#diff-28225bfdb29cfcfd891252f2c3fdc998f5acd86912756acae7780659b5009fb1).

It's currently used to help parse the `artifacts` and `boms` from `maven_install`.

Reply:
I'll wait for that PR to land, then I will reimplement this in top of it.

Reply:
The PR landed!